cli/src/semgrep/app/scans.py

```diff
@@ -48,7 +47,8 @@ class ScanCompleteResult:
 
 class ScanHandler:
     def __init__(self, dry_run: bool = False) -> None:
-        self.local_id = str(uuid4())
+        state = get_state()
```

Comment:
Please, avoid using get_state(). If you need some extra_info, just pass it down to the constructor.
Also like I said in the previous PR, it would be great to refactor ScanHandler to be built at ScanResponse 
so the self.scan_response can remove the Optional.

Reply:
Making that change requires rewriting the entire `semgrep CI` implementation (since not all scans use the scan handler, which was news to me recently). I don't have time for this at the moment since I lost a week to fixing all the various test tooling problems between the backend and the cli for the api changes I just made.

Reply:
I'd also add that I'm worried about the amount of change in any one release.